src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,105 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import { useState, useEffect } from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todos: Todo[];
```

Comment:
Don't pass the entire todos array here, just to filter it and return back. This is too complicated.

A better idea would be to move filtering logic to App.tsx. `TodoFilter`'s job is to select from `FilteredBy` enum